src/HDFFile.cpp

```diff
@@ -758,6 +765,30 @@ void HDFFile::createHDFStructures(
       if (Type == "dataset") {
         writeDataset(Parent, Value);
       }
+    } else if (auto TypeMaybe = find<std::string>("dtype", *Value)) {
+      auto Type = TypeMaybe.inner();
+      if (Type == "group") {
+        if (auto NameMaybe = find<std::string>("name", *Value)) {
+          auto Name = NameMaybe.inner();
+          try {
+            hdf_this = Parent.create_group(Name, LinkCreationPropertyList);
+            Path.push_back(Name);
+          } catch (...) {
+            LOG(Sev::Critical, "failed to create group  Name: {}", Name);
```

Comment:
Perhaps a discussion for another day, but I think we need to stop catching by ...
It seems unlikely to me that any library we use is going to throw something that doesn't inherit from std::exception and if it does we should put a wrapper around it so it doesn't propagate into our code.